pgmax/fg/nodes.py

```diff
@@ -99,17 +109,19 @@ def concatenate_wirings(wirings: Sequence) -> Wiring:
             "Please subclass the Wiring class and override this method."
         )
 
-    def compile_wiring(self, vars_to_starts: Mapping[Variable, int]) -> Wiring:
-        """Compile wiring for the factor
-
-        Args:
-            vars_to_starts: A dictionary that maps variables to their global starting indices
-                For an n-state variable, a global start index of m means the global indices
-                of its n variable states are m, m + 1, ..., m + n - 1
-
-        Returns:
-            Wiring for the Factor
-        """
-        raise NotImplementedError(
-            "Please subclass the Factor class and override this method"
-        )
+    # TODO: delete as this creates mypy issues
```

Comment:
What is the mypy issue?

Reply:
I get this
```
pgmax/factors/enumeration.py:160: error: Signature of "compile_wiring" incompatible with supertype "Factor"
pgmax/factors/enumeration.py:160: note:      Superclass:
pgmax/factors/enumeration.py:160: note:          @staticmethod
pgmax/factors/enumeration.py:160: note:          def compile_wiring(vars_to_starts: Mapping[Variable, int]) -> Wiring
pgmax/factors/enumeration.py:160: note:      Subclass:
pgmax/factors/enumeration.py:160: note:          @staticmethod
pgmax/factors/enumeration.py:160: note:          def compile_wiring(factor_edges_num_states: Any, variables_for_factors: Tuple[Variable, ...], factor_configs: Any, vars_to_starts: Mapping[Variable, int], num_factors: int) -> EnumerationWiring
pgmax/factors/logical.py:152: error: Signature of "compile_wiring" incompatible with supertype "Factor"
pgmax/factors/logical.py:152: note:      Superclass:
pgmax/factors/logical.py:152: note:          @staticmethod
pgmax/factors/logical.py:152: note:          def compile_wiring(vars_to_starts: Mapping[Variable, int]) -> Wiring
pgmax/factors/logical.py:152: note:      Subclass:
pgmax/factors/logical.py:152: note:          @staticmethod
pgmax/factors/logical.py:152: note:          def compile_wiring(factor_edges_num_states: Any, variables_for_factors: Tuple[Variable, ...], factor_sizes: Any, vars_to_starts: Mapping[Variable, int], edge_states_offset: int) -> LogicalWiring
Found 2 errors in 2 files (checked 31 source files)
```

Reply:
It indeed seems tricky to solve, so I added a check in `__post_init__`.